src/main/java/com/rackspace/salus/telemetry/ambassador/web/controller/AgentHistoryController.java

```diff
@@ -37,32 +36,29 @@
 public class AgentHistoryController {
 
   private final AgentHistoryService agentHistoryService;
-  private final AgentHistoryConversionService agentHistoryConversionService;
 
-  public AgentHistoryController(AgentHistoryService agentHistoryService,
-      AgentHistoryConversionService agentHistoryConversionService)  {
+  public AgentHistoryController(AgentHistoryService agentHistoryService)  {
     this.agentHistoryService = agentHistoryService;
-    this.agentHistoryConversionService = agentHistoryConversionService;
   }
 
   @GetMapping("/tenant/{tenantId}/agent-history")
   @ApiOperation(value = "Gets Agent History for Tenant")
-  public AgentHistoryOutput getAgentHistoryForTenant(
+  public PagedContent<AgentHistoryDTO> getAgentHistoryForTenant(
       @PathVariable String tenantId,
       @RequestParam(name = "envoyId", required = false) String envoyId,
-      @RequestParam(name = "resourceId", required = false) String resourceId)
+      @RequestParam(name = "resourceId", required = false) String resourceId,
+      Pageable pageable)
       throws NotFoundException {
-    Optional<AgentHistory> agentHistoryOptional = Optional.empty();
-    if(!StringUtils.isEmpty(envoyId))  {
-      agentHistoryOptional = agentHistoryService.getAgentHistoryForTenant(tenantId, envoyId);
+    if(!StringUtils.isEmpty(envoyId) && !StringUtils.isEmpty(resourceId)) {
+      throw new IllegalArgumentException("EnvoyId and ResourceId both cannot be not-null");
+    } else if(!StringUtils.isEmpty(envoyId))  {
+      return PagedContent.ofSingleton(new AgentHistoryDTO(agentHistoryService.getAgentHistoryForTenantAndEnvoyId(tenantId, envoyId)
+          .orElseThrow(() ->
+              new NotFoundException(String.format("No Agent History found for tenant %s and envoy %s ", tenantId, envoyId)))));
+    } else if(!StringUtils.isEmpty(resourceId))  {
+      return PagedContent.fromPage(agentHistoryService.getAgentHistoryForTenantAndResource(tenantId, resourceId, pageable)).map(AgentHistoryDTO::new);
```

Comment:
A couple of well placed newlines will reduce long lines like this and improve readability. Reviewing more of our PRs and code you will see some stylistic patterns, such as placing stream operators on a new line with its leading dot:

```suggestion
    return PagedContent.fromPage(
        agentHistoryService.getAgentHistoryForTenantAndResource(tenantId, resourceId, pageable)
    )
        .map(AgentHistoryDTO::new);
```